r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -239,6 +239,53 @@ test_that("str_replace and str_replace_all", {
 
 })
 
+test_that("strsplit and str_split", {
+  df <- tibble(x = c("Foo and bar", "baz and qux and quux"))
+  
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = strsplit(x, "and")) %>%
+      collect(),
+    df
+  )
+
+  expect_warning(
+    df %>%
+    Table$create() %>%
+      mutate(x = strsplit(x, "and.*", fixed = FALSE)) %>%
+      collect(),
+    regexp = "not supported in Arrow"
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      mutate(x = strsplit(x, "and.*", fixed = TRUE)) %>%
+      collect(),
+    df
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_split(x, "and")) %>%
+      collect(),
+    df
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_split(x, "and", n = 2)) %>%
+      collect(),
+    df
+  )
+
+  expect_warning(
+    df %>%
+      Table$create() %>%
+      mutate(x = str_split(x, "and.?")) %>%
+      collect()
+  )
+})
+
```

Comment:
Please add some tests here that exercise `str_split()` with the stringr pattern modifier functions:
- `fixed(pattern)` (should succeed)
- `regex(pattern)` with a non-regex pattern (should succeed)
- `regex(pattern)` with an actual regex pattern (should error)
- `fixed(pattern, ignore_case = TRUE)` with a non-regex pattern (should error because of the "Case-insensitive string splitting" code I added)

Reply:
Done now!